### ISSUE TITLE
Misleading compilation error when default font cannot be embedded when embedding glyphs

### ISSUE THREAD

**Issue by ogoffart** [slint-ui/slint#4997]
When building with slint_build::EmbedResourcesKind::EmbedForSoftwareRenderer, we get misleading compilation error when the default font don't exist.

To reproduce, edit the carousel_demo.slint and change the default font binding:
```slint
default-font-family: "NOT EXIST";
```

Then you get the error:

```
error: could not find font that provides specified family, falling back to Sans-Serif
    --> /home/olivier/slint/examples/carousel/ui/carousel_demo.slint:22:26
     |
  22 |     default-font-family: "NOT EXIST";
     |                          
```

It says it is falling back, but it doesn't since it errors out.

The error is even worse if the invalid font comes from the style.
Eg, modify the material-base/std-widget-impl.slint to change the default font 

```slint
    out property <string> default-font-family: "NOT EXIST";
```

and remove the default-font-family property from the window. 
Now we get:

```
error: internal error: fontdb could not determine a default font for sans-serif
   --> /home/olivier/slint/examples/carousel/ui/carousel_demo.slint:8:38
    |
  8 | export component MainWindow inherits Window {
    |                                      ^
```

**Comment by V0idMatr1x** [comment]
I am also having this issue on FreeBSD... Anyone have a fix yet? I used the default template with cargo-generate. It seemed to build fine, but when I ran `cargo run` I got the error you see below.

![image](https://github.com/slint-ui/slint/assets/88049272/288fd7ff-f749-4ffe-84d5-74623d4e9db2)

**Comment by tronical** [comment]
Similar issue indeed. Do you have libfontconfig on your system installed?

**Comment by V0idMatr1x** [comment]
> Similar issue indeed. Do you have libfontconfig on your system installed?

Yes, it was installed. That was the first thing I thought to check. Thanks for the quick response!

**Comment by tronical** [comment]
Ok - next thing to check :). What does `fc-match sans-serif` print?

**Comment by V0idMatr1x** [comment]
> Ok - next thing to check :). What does `fc-match sans-serif` print?

It prints `NotoSans-Regular.ttf: "Noto Sans" "Regular"`
